Calibration/HcalAlCaRecoProducers/src/AlCaGammaJetProducer.cc

```diff
@@ -70,25 +66,41 @@ class AlCaGammaJetProducer : public edm::EDProducer {
+  edm::EDPutTokenT<edm::TriggerResults> put_trigger_;
+  edm::EDPutTokenT<std::vector<Bool_t>> put_loosePhot_;
+  edm::EDPutTokenT<std::vector<Bool_t>> put_tightPhot_;
+  edm::EDPutTokenT<double> put_rho_;
+  edm::EDPutTokenT<reco::PFCandidateCollection> put_pfCandidate_;
+  edm::EDPutTokenT<reco::VertexCollection> put_vertex_;
+  edm::EDPutTokenT<reco::PFMETCollection> put_pfMET_;
+  edm::EDPutTokenT<reco::GsfElectronCollection> put_gsfEle_;
+  edm::EDPutTokenT<reco::ConversionCollection> put_conv_;
+  edm::EDPutTokenT<reco::BeamSpot> put_beamSpot_;
 };
 
-AlCaGammaJetProducer::AlCaGammaJetProducer(const edm::ParameterSet& iConfig) : nAll_(0), nSelect_(0) {
+AlCaGammaJetProducer::AlCaGammaJetProducer(const edm::ParameterSet& iConfig) {
   // Take input
-  labelPhoton_ = iConfig.getParameter<edm::InputTag>("PhoInput");
-  labelPFJet_ = iConfig.getParameter<edm::InputTag>("PFjetInput");
-  labelHBHE_ = iConfig.getParameter<edm::InputTag>("HBHEInput");
-  labelHF_ = iConfig.getParameter<edm::InputTag>("HFInput");
-  labelHO_ = iConfig.getParameter<edm::InputTag>("HOInput");
-  labelTrigger_ = iConfig.getParameter<edm::InputTag>("TriggerResults");
-  labelPFCandidate_ = iConfig.getParameter<edm::InputTag>("particleFlowInput");
-  labelVertex_ = iConfig.getParameter<edm::InputTag>("VertexInput");
-  labelPFMET_ = iConfig.getParameter<edm::InputTag>("METInput");
-  labelGsfEle_ = iConfig.getParameter<edm::InputTag>("gsfeleInput");
-  labelRho_ = iConfig.getParameter<edm::InputTag>("rhoInput");
-  labelConv_ = iConfig.getParameter<edm::InputTag>("ConversionsInput");
-  labelBeamSpot_ = iConfig.getParameter<edm::InputTag>("BeamSpotInput");
-  labelLoosePhot_ = iConfig.getParameter<edm::InputTag>("PhoLoose");
-  labelTightPhot_ = iConfig.getParameter<edm::InputTag>("PhoTight");
+  auto labelPhoton_ = iConfig.getParameter<edm::InputTag>("PhoInput");
+  auto labelPFJet_ = iConfig.getParameter<edm::InputTag>("PFjetInput");
+  auto labelHBHE_ = iConfig.getParameter<edm::InputTag>("HBHEInput");
+  auto labelHF_ = iConfig.getParameter<edm::InputTag>("HFInput");
+  auto labelHO_ = iConfig.getParameter<edm::InputTag>("HOInput");
+  auto labelTrigger_ = iConfig.getParameter<edm::InputTag>("TriggerResults");
+  auto labelPFCandidate_ = iConfig.getParameter<edm::InputTag>("particleFlowInput");
+  auto labelVertex_ = iConfig.getParameter<edm::InputTag>("VertexInput");
+  auto labelPFMET_ = iConfig.getParameter<edm::InputTag>("METInput");
+  auto labelGsfEle_ = iConfig.getParameter<edm::InputTag>("gsfeleInput");
+  auto labelRho_ = iConfig.getParameter<edm::InputTag>("rhoInput");
+  auto labelConv_ = iConfig.getParameter<edm::InputTag>("ConversionsInput");
+  auto labelBeamSpot_ = iConfig.getParameter<edm::InputTag>("BeamSpotInput");
+  auto labelLoosePhot_ = iConfig.getParameter<edm::InputTag>("PhoLoose");
+  auto labelTightPhot_ = iConfig.getParameter<edm::InputTag>("PhoTight");
```

Comment:
I don't really get your consideration of creating new objects here. Could you offer some explanation?

Reply:
Thanks for the catch! This was a left over from an early attempt to see if I could do away with the InputTags as member data. I'll fix that.